### ISSUE TITLE
BelongsToMany Ignores bindingKey

### ISSUE THREAD

**Issue by shayllis** [cakephp/cakephp#12238]
* [x] bug
* [x] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.5.13
* Platform and Target: MacOS, localhost (AJAX request), MySQL Database

### What you did
Set up BelongsToMany relation without a joining class

### What happened
Relation was created without my given field for bindingKey

### What you expected to happen
Association between my tables to use my given bindingKey

### My Solution
As you can set bindingKey, CakePHP does not it at all. It turned that I could (and should) fix it myself.
He is what I have done:

I updated every junctionGeneration method under "Core/ORM/Association/BelongsToMany".
See:

```
// Core/ORM/Association/BelongsToMany
// Implemented method
public function getBindingKey()
{
     if ($this->_bindingKey === null) {
         $this->_bindingKey = $this->_modelKey($this->getSource()->getTable());
     }

     return $this->_bindingKey;
}
// Changed method
protected function _generateJunctionAssociations($junction, $source, $target)
    {
        $tAlias = $target->getAlias();
        $sAlias = $source->getAlias();

        if (!$junction->association($tAlias)) {
            $junction->belongsTo($tAlias, [
                'foreignKey' => $this->getTargetForeignKey(),
	        'bindingKey' => $this->getBindingKey(), // New line
                'targetTable' => $target
            ]);
        }
        if (!$junction->association($sAlias)) {
            $junction->belongsTo($sAlias, [
                'foreignKey' => $this->getForeignKey(),
                'bindingKey' => $this->getBindingKey(), // New line
                'targetTable' => $source
            ]);
        }
    }

```

**Comment by lorenzo** [comment]
I think we need a different name for this as `bindingKey` is already used for something else, and probably we would need one for the `sourceTable` and another for the `targetTable`.

Do you think you can provide a pull request for this new feature?

**Comment by shayllis** [comment]
Sure: [https://github.com/cakephp/cakephp/pull/12239](https://github.com/cakephp/cakephp/pull/12239).

BindingKey is also a name given in belongsTo Associations, as you said, to set origin table key. I think bindingKey could be used in a belongsToMany because its referring to the current table. To avoid confusing, it could be given another name as well. I suggest "currentTableKey".

**Comment by markstory** [comment]
Refs #12061

**Comment by markstory** [comment]
This will be fixed in 4.1
